src/bindgen/bindings.rs

```diff
@@ -204,151 +200,46 @@ impl Bindings {
-                        out.new_line();
-                    }
-                    out.write("#include <stdint.h>");
-                    out.new_line();
-                    out.write("#include <stdlib.h>");
-                    out.new_line();
-                }
-                Language::Cxx => {
-                    out.write("#include <cstdarg>");
-                    out.new_line();
-                    if self.config.usize_is_size_t {
-                        out.write("#include <cstddef>");
-                        out.new_line();
-                    }
-                    out.write("#include <cstdint>");
-                    out.new_line();
-                    out.write("#include <cstdlib>");
-                    out.new_line();
-                    out.write("#include <ostream>");
-                    out.new_line();
-                    out.write("#include <new>");
-                    out.new_line();
-                    if self.config.enumeration.cast_assert_name.is_none()
-                        && (self.config.enumeration.derive_mut_casts
-                            || self.config.enumeration.derive_const_casts)
-                    {
-                        out.write("#include <cassert>");
-                        out.new_line();
-                    }
-                }
-                Language::Cython => {
-                    out.write(
-                        "from libc.stdint cimport int8_t, int16_t, int32_t, int64_t, intptr_t",
-                    );
-                    out.new_line();
-                    out.write(
-                        "from libc.stdint cimport uint8_t, uint16_t, uint32_t, uint64_t, uintptr_t",
-                    );
-                    out.new_line();
-                    out.write("cdef extern from *");
-                    out.open_brace();
-                    out.write("ctypedef bint bool");
-                    out.new_line();
-                    out.write("ctypedef struct va_list");
-                    out.new_line();
-                    out.close_brace(false);
-                }
+    pub fn write<F: Write>(&self, file: F) {
+        match self.config.language {
+            Language::Cxx | Language::C => {
+                self.write_with_backend(file, &CLikeLanguageBackend::new(self.config.clone()))
             }
-        }
-
-        for include in self.config.sys_includes() {
-            write!(out, "#include <{}>", include);
-            out.new_line();
-        }
-
-        for include in self.config.includes() {
-            write!(out, "#include \"{}\"", include);
-            out.new_line();
-        }
-
-        if self.config.language == Language::Cython {
-            for (module, names) in &self.config.cython.cimports {
-                write!(out, "from {} cimport {}", module, names.join(", "));
-                out.new_line();
+            Language::Cython => {
+                self.write_with_backend(file, &CythonLanguageBackend::new(self.config.clone()))
```

Comment:
Cloning the config here seems kind of unfortunate? Ideally the backend is an empty struct and the config is either given by callers or part of the source writer?

Reply:
I managed to avoid the clone by only passing a reference (we only need the config read only and the language backend is dropped after this call)